### PR TITLE
Fix function tracing for New Relic Agent 5.0.0.124

### PR DESCRIPTION
**Pull request by ormsbee**
Version 5.0.0.124 of the New Relic Agent changed the function tracing
API. Using it the old way was resulting in many thousands of new metrics
being added (each process invocation of a function trace was creating a
new metric name), so we ended up with a bunch of "metrics" named like:

determine_group_permissions/<newrelic.api.web_transaction.WSGIWebTransaction object at 0x7f09bf016ad0>
determine_group_permissions/<newrelic.api.web_transaction.WSGIWebTransaction object at 0x7f09bf098510>
determine_group_permissions/<newrelic.api.web_transaction.WSGIWebTransaction object at 0x7f09bf0a8bd0>
determine_group_permissions/<newrelic.api.web_transaction.WSGIWebTransaction object at 0x7f09bf108b50>
determine_group_permissions/<newrelic.api.web_transaction.WSGIWebTransaction object at 0x7f09bf112250>

See: https://docs.newrelic.com/docs/release-notes/agent-release-notes/python-release-notes/python-agent-500124

**Description:**

Describe in a couple of sentence what this PR adds

**JIRA:**

[XXX-XXXX](https://openedx.atlassian.net/browse/XXX-XXXX)

**Dependencies:**

List dependencies on other outstanding PRs, issues, etc.

**Merge deadline:**

List merge deadline (if any)

**Installation instructions:**

List any non-trivial installation
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Reviewers:**
- [ ] @edx/arch-review
- [ ] tag reviewer

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed
- [ ] PR author is listed in AUTHORS

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPi after tag-triggered build is 
      finished.
- [ ] Delete working branch (if not needed anymore)

**Author concerns:**

List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

### CODE REVIEW

**Comment by ormsbee** [comment]
It's not clear to me how to write a test that is actually meaningful here. Suggestions welcome. :-/

**Comment by nedbat** [comment]
Wow, super-crappy API design. Change the signature, but silently accept the old arguments and use them to bork the monitoring.

If we had to have a test, it would mock out new relic, report a version, and check the mock call was correct for that version.  But I don't think it's worth it.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,15 +6,32 @@ Change Log
...
-   
+
    This project adheres to Semantic Versioning (http://semver.org/).
 
 .. There should always be an "Unreleased" section for changes pending release.
 
-Unreleased
-~~~~~~~~~~
+-Unreleased
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (nedbat) [inline CHANGELOG.rst]: "-Unreleased" ?  Also, why are the rules longer than the text they underline?

Comment (ormsbee) [reply CHANGELOG.rst]: Copy paste fail on my part. Will fix.

**Comment by ormsbee** [comment]
@robrap: How much do we care about the code coverage loss? I could mock it with a test, but it would seriously be creating a mock exactly for this call, and wouldn't actually guard against regressions in the real underlying lib...?

**Pull request merged**